### PR TITLE
libarchive/3.6.x: Fix compile for Android

### PR DESCRIPTION
**Pull request by bog-dan-ro**
Bump version in the process

Specify library name and version:  **libarchive/3.6.x**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libarchive/all/patches/android_compile.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ #
+ ############################################
+ 
++if (ANDROID)
++    include_directories(${PROJECT_SOURCE_DIR}/contrib/android/include)
++endif()
++
+ # Public headers
+ SET(include_HEADERS
+   archive.h
```
[the remark this answers is not in the document]
Comment (bog-dan-ro) [reply recipes/libarchive/all/patches/android_compile.patch:15]: yup, though libarchive seems a bit unmaintained ...

[the remark this answers is not in the document]
Comment (bog-dan-ro) [reply recipes/libarchive/all/patches/android_compile.patch:15]: > Is is a fix for [libarchive/libarchive#890](https://github.com/libarchive/libarchive/issues/890)? Could you submit a PR upstream?

Done https://github.com/libarchive/libarchive/pull/1729

**Review by uilianries — APPROVED** [review]

**Comment by bog-dan-ro** [comment]
Is it my fault? I can't see the error ...

**Comment by uilianries** [comment]
@bog-dan-ro Nope. Just need a CI restart ... I'm doing it now.

**Comment by bog-dan-ro** [comment]
> @bog-dan-ro Nope. Just need a CI restart ... I'm doing it now.

Do I need to do anything else or it will reschedule the build itself?

**Comment by uilianries** [comment]
@bog-dan-ro You just need to wait, the `continuous-integration/jenkins/pr-merge Pending — This commit is being built ` is marker to say that's is running. However, we are restarting more builds, so it may take few hours.

**Comment by uilianries** [comment]
finally passed!

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**